api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/Contributor.java

```diff
@@ -11,10 +11,26 @@
 /**
  * Support loading an Instance of a service, optionally providing it with configuration obtained
  * from the Kroxylicious configuration file.
+ *
+ * @param <T> the service type
  */
 public interface Contributor<T> {
 
+    /**
+     * Gets the concrete type of the configuration required by this service instance.
+     * @param shortName service short name
+     *
+     * @return class of a concrete type or null, if this service instance does not accept configuration.
+     */
     Class<? extends BaseConfig> getConfigType(String shortName);
 
+    /**
+     * Creates a instance of the service.
```

Comment:
For my money `creates a instance` implies a different contract to the method name `getInstance`. The method name implies caching/singletons are supported however `creates a instance` implies that every invocation gets a new instance. 

I'm not sure which contract we should prefer or mean to imply but I think we should be consistent between the javadoc and method names.

Also while I'm here in the grammar pedant category shouldn't it be 
```suggestion
     * Creates an instance of the service.
```

Reply:
While https://github.com/kroxylicious/kroxylicious/commit/e47f523d8fbeadfcfcc910989ef77de850ed6ffd fixes the nit. It doesn't address the wider point I was making about the consistency between the doc and the method name.

Reply:
Good point.  I've opted to rename the method.  I think the simpler semantics are good enough for now.  If we took the alternative path, we'd need to think about the immutability of the `getInstance()'s` parameters and potentially lifecycling of the service instance.

Reply:
Yep, sounds good.